karapace/kafka_rest_apis/__init__.py

```diff
@@ -398,7 +398,11 @@ async def partition_publish(self, topic, partition_id, content_type, *, request)
 
     async def topic_publish(self, topic: str, content_type: str, *, request):
         self.log.debug("Executing topic publish on topic %s", topic)
-        await self.publish(topic, None, content_type, request.content_type, request.json)
+        body = request.json
+        if not isinstance(body, dict):
+            raise HTTPResponse(body="Request body is not JSON object", status=HTTPStatus.BAD_REQUEST)
```

Comment:
I think it makes sense to add validation of the request body with a schema using something like pydantic, which gives more precise errors.

Reply:
@tvainika could you please add an update here.

Reply:
Makes sense to use pydantic instead of this.